lib/puppet/indirector/file_server.rb

```diff
@@ -15,7 +15,15 @@ def authorized?(request)
 
     # If we're not serving this mount, then access is denied.
     return false unless mount
-    mount.allowed?(request.node, request.ip)
+
+    # If there are no auth directives or there is an 'allow *' directive, then
+    # access is allowed.
+    if mount.empty? || mount.globalallow?
```

Comment:
As I mentioned in a [comment](https://tickets.puppetlabs.com/browse/PUP-6359?focusedCommentId=449247&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-449247) on the related JIRA ticket, I was a little on the fence about whether it's best to grant access when there are no security directives present in the mount point definition - where `mount.empty?` would be true - because on upgrade it would be possible for authentication to be granted to a mount point for which it had previously been denied.

Ultimately, though, I think this is best assuming that we'd at some point like to eliminate the presence / handling of security directives from the mount point definitions completely - presumably at the same time we would gut support for Ruby auth.conf from the Puppet code.